src/resource.rs

```diff
@@ -2,35 +2,325 @@
 //!
 //! TODO: YAML declaration and parsing. This should be similar to a plugin
 //! system (outside the `types` module), which also parses nested [`OpDef`]s.
-//!
-//! [`OpDef`]: crate::ops::custom::OpDef
 
-use std::collections::HashSet;
-use std::fmt::{self, Display};
+use serde::{Deserializer, Serializer};
+use std::collections::hash_map::Entry;
+use std::collections::{HashMap, HashSet};
+use std::fmt::{Debug, Display, Formatter};
+use std::sync::Arc;
 
 use smol_str::SmolStr;
+use thiserror::Error;
+
+use crate::types::type_param::check_arg;
+use crate::types::TypeRow;
+use crate::types::{
+    type_param::{TypeArg, TypeParam},
+    Signature, SignatureDescription,
+};
+use crate::Hugr;
+
+/// Trait for resources to provide custom binary code for computing signature.
+pub trait CustomSignatureFunc: Send + Sync {
+    /// Compute signature of node given the operation name,
+    /// values for the type parameters,
+    /// and 'misc' data from the resource definition YAML
+    fn compute_signature(
+        &self,
+        name: &SmolStr,
+        arg_values: &[TypeArg],
+        misc: &HashMap<String, serde_yaml::Value>,
+    ) -> Result<(TypeRow, TypeRow, ResourceSet), SignatureError>;
```

Comment:
Well, firstly there's the question of whether external/resource ops should be able to have `static_inputs` - currently the belief is, no they should not be.

Secondly there's the `input_resources` question, and when that's resolved (i.e. Signature stores the delta) this will be a stage closer to a Signature. But I don't think we intend to allow const inputs.

So, post #189, we could end up with something like
```
struct GraphType {inputs: TypeRow, outputs: TypeRow, resource_delta: ResourceSet};
struct Signature { ty: GraphType, static_inputs: TypeRow }
```
but not there yet.

Reply:
Agustin showed me how to do this, admittedly it requires enabling a feature of Serde globally (which we might not want everywhere), but if we can live with that then the rest is easy :)